### PR TITLE
SONARJS-946 Improve ImmediatelyReturnedVariableCheck: ignore if variable is used

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-checks/src/test/resources/checks/ImmediatelyReturnedVariable.js [hunk 1]
```diff
@@ -118,3 +118,11 @@ var arrow_function_ok = (a, b) => {
 }
 
 var arrow_function_no_block = (a, b) =>  a + b;
+
```
Comment (ivandalbosco) [inline javascript-checks/src/test/resources/checks/ImmediatelyReturnedVariable.js]: @vilchik-elena Your implementation is very robust! How about adding a unit test with a wrapping array:
```
function variable_is_used() {
  const bar = [             // OK
     {
       doSomethingElse(p) { },
       doSomething() { bar.doSomethingElse(1); }
     },
     "hello" ];
  return bar;
}
```

Comment (ivandalbosco) [inline javascript-checks/src/test/resources/checks/ImmediatelyReturnedVariable.js]: @vilchik-elena I also suggest a unit test with the usage of a homonymous local variable "bar":
```
function homonymous_is_used() {
  const bar = {           // OK
     doSomethingElse(p) { var bar = 2; return p + bar; },
     doSomething() { return doSomethingElse(1); }
  };
  return bar;
}
```

File: its/ruling/src/test/expected/javascript-S1488.json [hunk 2]
```diff
@@ -190,7 +156,6 @@
 'project:dojo-1.8.1/dojox/jq.js.uncompressed.js':[
 706,
 785,
-1755,
```
Comment (ivandalbosco) [inline its/ruling/src/test/expected/javascript-S1488.json]: @elena Isn't this is new FN?
```
var dfd = dojo.io.script.get(args);
return dfd;
```

File: its/ruling/src/test/expected/javascript-S1488.json [hunk 3]
```diff
@@ -273,9 +235,6 @@
 'project:ecmascript6/sonar-web/src/main/js/apps/coding-rules/facets/status-facet.js':[
 9,
 ],
-'project:ecmascript6/watchtower.js/test/watchgroup.spec.js':[
-593,
```
Comment (ivandalbosco) [inline its/ruling/src/test/expected/javascript-S1488.json]: @vilchik-elena New FN?

File: its/ruling/src/test/expected/javascript-S1488.json (hunk 2 above)
Comment (vilchik-elena) [inline its/ruling/src/test/expected/javascript-S1488.json]: @ivandalbosco fixed

File: its/ruling/src/test/expected/javascript-S1488.json (hunk 3 above)
Comment (vilchik-elena) [inline its/ruling/src/test/expected/javascript-S1488.json]: @ivandalbosco same nature as previous one. Thanks for noticing!

File: javascript-checks/src/test/resources/checks/ImmediatelyReturnedVariable.js (hunk 1 above)
Comment (vilchik-elena) [inline javascript-checks/src/test/resources/checks/ImmediatelyReturnedVariable.js]: @ivandalbosco I added second one, but first one is IMO redundant

**Pull request merged**